osquery/sql/sqlite_util.cpp

```diff
@@ -459,41 +460,109 @@ Status QueryPlanner::applyTypes(TableColumns& columns) {
-                << " in query result consider using aliases";
+  do {
+    RowTyped row;
+    for (int i = 0; i < num_columns; i++) {
+      switch (sqlite3_column_type(prepared_statement, i)) {
+      case SQLITE_INTEGER:
+        row[colNames[i]] =
+            static_cast<int64_t>(sqlite3_column_int64(prepared_statement, i));
+        break;
+      case SQLITE_FLOAT:
+        row[colNames[i]] = sqlite3_column_double(prepared_statement, i);
+        break;
+      case SQLITE_NULL:
+        row[colNames[i]] = FLAGS_nullvalue;
+        break;
+      default:
+        // Everything else (SQLITE_TEXT, SQLITE3_TEXT, SQLITE_BLOB) is
+        // obtained/conveyed as text/string
+        row[colNames[i]] = std::string(reinterpret_cast<const char*>(
+            sqlite3_column_text(prepared_statement, i)));
       }
-      r[column[i]] = (argv[i] != nullptr) ? argv[i] : FLAGS_nullvalue;
     }
-  }
-  (*qData).push_back(std::move(r));
-  return 0;
+    results.push_back(std::move(row));
+    rc = sqlite3_step(prepared_statement);
+  } while (SQLITE_ROW == rc);
+  return rc;
 }
 
-Status queryInternal(const std::string& q,
-                     QueryData& results,
+Status queryInternal(const std::string& query,
+                     QueryDataTyped& results,
                      const SQLiteDBInstanceRef& instance) {
-  char* err = nullptr;
-  auto lock = instance->attachLock();
-  sqlite3_exec(instance->db(), q.c_str(), queryDataCallback, &results, &err);
+  sqlite3_stmt* prepared_statement{nullptr}; /* Statement to execute. */
+
+  int rc = SQLITE_OK; /* Return Code */
+  const char* leftover_sql = nullptr; /* Tail of unprocessed SQL */
+  const char* sql = query.c_str(); /* SQL to be processed */
+
+  /* The big while loop.  One iteration per statement */
+  while ((sql[0] != 0) && (SQLITE_OK == rc)) {
+    const auto lock = instance->attachLock();
+
+    rc = sqlite3_prepare_v2(
+        instance->db(), sql, -1, &prepared_statement, &leftover_sql);
+    if (rc != SQLITE_OK || prepared_statement == nullptr) {
+      if (prepared_statement != nullptr) {
+        sqlite3_finalize(prepared_statement);
+      }
+      return Status::failure(sqlite3_errmsg(instance->db()));
+    } else {
+      if (prepared_statement == nullptr) {
+        /* this happens for a comment or white-space */
+        sql = leftover_sql;
+        while (isspace(sql[0])) {
+          sql++;
+        }
+        continue;
+      }
+
+      rc = sqlite3_step(prepared_statement);
+      /* if we have a result set row... */
+      if (SQLITE_ROW == rc) {
+        rc = readRows(prepared_statement, results);
+      }
+
+      /* Finalize the statement just executed, regardless.  */
+      int rc2 = sqlite3_finalize(prepared_statement);
+      /* Return error if rc of or just prior to finalize is not OK */
+      if (!(rc == SQLITE_OK || rc == SQLITE_DONE) || rc2 != SQLITE_OK) {
+        return Status(1, sqlite3_errmsg(instance->db()));
```

Comment:
Status::failure(msg)

Reply:
Doh!  @akindyakov said the same.  :)

Reply:
It's kind of old style, we should remove this at some point. :(

Reply:
I changed it (I'd just forgotten to....or actually changed, reverted the file, then forgot to re-change).